File: redsheet-portal/src/app/database-models/user-message.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { User,  } from './'
export class UserMessage {
    constructor(userMessage?: UserMessage )  {
        let properties = [ "createdOn", "message", "updatedOn", "userMessageId", "createdBy", "statusId", "updatedBy",];
        for (let property of properties)
            if (userMessage && userMessage[property])
                this[property] = userMessage[property];
    }
 
	createdOn : Date =   undefined;
 
    @required()
	message : string =   undefined;
 
	updatedOn : Date =   undefined;
 
	userMessageId : number =   0 ;
 
    @range(0,2147483647)
	createdBy : number =   undefined;
	user : User  ;
 
	statusId : number =   undefined;
 
	updatedBy : number =   undefined;
	user1 : User  ;


}

File: RedSheet.DbEntities/Models/UserMessage.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("UserMessages", Schema = "dbo")]
	[Serializable]
	public partial class UserMessage  
	{

	public Nullable<System.DateTime> CreatedOn { get; set; } 
 
	[Required]
	public string Message { get; set; } 

	public Nullable<System.DateTime> UpdatedOn { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int UserMessageId { get; set; } 
 
	[RelationshipTableAttribue("Users","dbo")]
	//Relationships
	public int CreatedBy { get; set; } 
 
	[ForeignKey("CreatedBy")]
	public User User { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Nullable<Status> StatusId { get; set; } 
 
	[RelationshipTableAttribue("Users","dbo")]
	//Relationships
	public Nullable<int> UpdatedBy { get; set; } 
 
	[ForeignKey("UpdatedBy")]
	public User User1 { get; set; } 

	public UserMessage()
	{
    }

    }
}

